ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/rename/AbstractAlterTableRenameAnalyzer.java

```diff
@@ -49,6 +55,13 @@ protected void analyzeCommand(TableName tableName, Map<String, String> partition
       setAcidDdlDesc(desc);
     }
     addInputsOutputsAlterTable(tableName, null, desc, desc.getType(), false);
+//    inputs.add(new ReadEntity(table));
+    outputs.clear();
```

Comment:
doing `outputs.clear()` right after `addInputsOutputsAlterTable` is pretty strange

could you please put this code into that method - isn't there where it supposed to live?

Reply:
Yeah, I had to do this here. Reason: When a user executes Alter table foo rename to otherdatabase.foo2; we never send the otherdatabase and new table name (foo2) objects for authorization. We need to check whether the end-user has privileges to create a table in otherdatabase. So I think this special logic has to be in AbstractTableRenameAnalyzer instead of AbstractTableAnalyzer.

Reply:
I see - but this will make it more diverse...what do you think about the following:
* split the `addInputsOutputsAlterTable` into 2 methods which are taking care of inputs/outputs
* you could add back `addInputsOutputsAlterTable` which is calling these 2 for backward compatibility
* and call these 2 methods with old/new args from here?